pathfinder/pathfinder.py

```diff
@@ -0,0 +1,107 @@
+from pathfinder.city import City
+from pathfinder.graphs import Graph
+from pathfinder.types import Path, CityInfo
+
+
+class PathFinder:
+
+    def __init__(self, graph: Graph):
+        self._graph = graph
+        self._cities: dict[City, CityInfo] = {}
+
+    def get_shortest_path(self, start: City, end: City) -> Path:
+        """Returns the shortest path to the end city"""
+
+        self._init_cities_dict(start)
+
+        current_cities: list[City] = [start]
+        next_cities_to_visit: list[City] = [start]
+        visited_cities: list[City] = []
+
+        while len(current_cities) > 0:
+
+            current_cities = next_cities_to_visit.copy()
+            next_cities_to_visit.clear()
+
+            for current_city in current_cities:
```

Comment:
Cette boucle est en trop : on ne souhaite pas itérer sur les villes à visiter. On souhaite en choisir une (tant qu'il y a en a, c'est assuré par le while ligne 21). 
Un print te montrera les trop nombreux passages dans ce for.